### PR TITLE
Misc. web code updates

### PR DESCRIPTION
**Pull request by brianhelba**

### CODE REVIEW

File: web_external/Studies/CreateStudyView.js [hunk 1]
```diff
@@ -151,18 +151,20 @@ isic.views.CreateStudyView = isic.View.extend({
             },
             error: null
         }).done(_.bind(function () {
+            var escapedName = this.$('<span>').text(name).html();
```
Comment (msmolens) [inline web_external/Studies/CreateStudyView.js]: Would `_.escape()` accomplish the goal in a cleaner way?

File: web_external/model.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    }
+});
+
+Backbone.ajax = function (options) {
+    // girder.restRequest expects a "path" option, and will set "url" internally
+    var params = _.clone(options);
+    params.path = params.url;
+    delete params.url;
+
+    return girder.restRequest(params);
```
Comment (msmolens) [inline web_external/model.js]: This gives me `Maximum call stack size exceeded` because `girder.restRequest()` calls `Backbone.ajax()`.

Comment (brianhelba) [inline web_external/model.js]: Oops, I lost the fix for that in my rebase. I'll fix again on this branch.

File: web_external/Studies/CreateStudyView.js (hunk 1 above)
Comment (brianhelba) [inline web_external/Studies/CreateStudyView.js]: I didn't see that guy. Nice find, thanks.

File: web_external/model.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        return this.resourceName;
+    }
+});
+
+Backbone.sync = function (method, model, options) {
+    // Reimplement "Backbone.sync" to use "girder.restRequest" in place of
+    // "Backbone.ajax". "Backbone.ajax" cannot be directly overridden, as it
+    // is called by "girder.restRequest". The only important changes here are:
+    //   * Use "girder.restRequest" instead of "Backbone.ajax"
+    //   * Set "params.path" instead of "params.url"
```
Comment (msmolens) [inline web_external/model.js]: Can we add a comment explaining why we reimplement `Backbone.sync`?

Comment (brianhelba) [inline web_external/model.js]: Done.

File: web_external/model.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+isic.Model = girder.Model.extend({
+    urlRoot: function () {
+        return this.resourceName;
+    }
+});
+
+Backbone.sync = function (method, model, options) {
+    // In order to use the native "Backbone.destroy" method (which triggers the correct
```
Comment (msmolens) [inline web_external/model.js]: Should that be "Backbone.Model.destroy"?

File: web_external/model.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+isic.Model = girder.Model.extend({
+    urlRoot: function () {
+        return this.resourceName;
+    }
+});
+
+Backbone.sync = function (method, model, options) {
```
Comment (msmolens) [inline web_external/model.js]: This seems general enough to go in `init.js`, maybe?

Comment (brianhelba) [inline web_external/model.js]: Good point. Done.

**Pull request merged**